### PR TITLE
Add postsubmit check for macOS

### PR DESCRIPTION
**Pull request by antiagainst**
This commit add postsubmit checks for macOS. It uses our
own self-hosted Mac Mini to run the jobs.

### CODE REVIEW

**Comment by antiagainst** [comment]
It runs successfully after manually triggered: https://github.com/openxla/iree/actions/runs/4327020994.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -252,6 +252,39 @@ jobs:
...
+      - ${{ github.repository == 'openxla/iree' && 'self-hosted' || 'macos-11' }}  # must come first
+      - runner-group=postsubmit
+      - os-family=macOS
+    env:
+      BUILD_DIR: build-macos
+    steps:
+      - name: "Checking out repository"
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: "Updating git submodules"
+        run: git submodule update --init --jobs 8 --depth 1
```
Comment (ScottTodd) [inline .github/workflows/ci.yml:268]: Does this explicit submodule fetch outperform 

```yaml
    steps:
      - name: "Checking out repository"
        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
        with:
          submodules: true
```
? (On Windows we saw poor behavior in the checkout action and wanted to ensure we were doing just what we wanted)

**Review by GMNGeoffrey — APPROVED** [review]

**Review by ScottTodd — COMMENTED** [review]
Nice!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (antiagainst) [reply .github/workflows/ci.yml:268]: Good question. I just copied from what you have in the Windows session there. :) It takes 2s to checkout the repo and 34s to fetch submodules this way as shown in https://github.com/openxla/iree/actions/runs/4327020994/jobs/7555217216. Let me try it out the above.

Comment (antiagainst) [reply .github/workflows/ci.yml:268]: Using the above takes 1min30s: https://github.com/openxla/iree/actions/runs/4327186867/jobs/7555521072. So yeah explicit submodule checkout saves ~1min.

**Review by ScottTodd — APPROVED** [review]

**Pull request merged**